### DISCUSSION TITLE
Changing Lua to use global functions rather than member functions

### DISCUSSION THREAD

**Discussion by cldellow** [systemed/tilemaker#587] [General]
This has the potential to be disruptive to users, so I figured I'd start a discussion here before making a PR against Tilemaker proper.

https://github.com/cldellow/tilemaker/pull/1 changes how the Lua interop works, reducing read time by 7.5% (wall clock) or 11.5% (user time)

Would you be open to this change? I can see how the current approach would feel quite familiar to folks and provides a nice namespacing benefit.

If you were open to it, would you prefer it against master or refactor_geometries?

Argument in favour of master: The nature of the change is that it will create ongoing merge conflicts in process.lua, so it'll make merging master into refactor_geometries more painful.

Argument in favour of refactor_geometries: it's a breaking change to something user-facing.

**Comment by systemed** [comment]
That's pretty neat and a worthwhile saving, especially when scaling up to the planet. I'd be happy to go for that but I think definitely `refactor_geometries` - it's the sort of change that requires a major version increment, and `refactor_geometries` is intended to be tilemaker 3.0 so it would make sense there. The Lua conflicts should be fairly easy to handle I think.

**Reply by cldellow** [reply]
Sounds good! I'll send a PR.
